### PR TITLE
WT-9684 improve stability in latest IDE

### PR DESCRIPTION
**Pull request by sukolenvo**

### CODE REVIEW

File: src/main/java/io/wisetime/idea/branch/WisetimeFrameTitleManager.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
+
+public class WisetimeFrameTitleManager extends PlatformFrameTitleBuilder {
+
+  @Override
+  public String getProjectTitle(@NotNull Project project) {
+    final String currentBranch = Optional.ofNullable(ServiceManager.getService(project, BranchHelper.class))
+        .map(BranchHelper::getCurrentBranchName)
+        .orElse(null);
```
Comment (tom-haines) [inline src/main/java/io/wisetime/idea/branch/WisetimeFrameTitleManager.java]:
```suggestion
        // return empty string if branch name is unavailable
        .orElse("");
```

File: src/main/java/io/wisetime/idea/branch/WisetimeFrameTitleManager.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    return getProjectTitleWithBranch(project, currentBranch);
+  }
+
+  String getProjectTitleWithBranch(Project project, String branchName) {
+    String projectTitle = super.getProjectTitle(project);
+    if (branchName != null && !branchName.isEmpty()) {
+      projectTitle = projectTitle.replaceAll("^(.*)\\[.*]$", "$1[" + branchName + "]");
```
Comment (tom-haines) [inline src/main/java/io/wisetime/idea/branch/WisetimeFrameTitleManager.java:24]:
```suggestion
      // if an existing title ends with context data inside square brackets, replace with the update updated branch title
      projectTitle = projectTitle.replaceAll("^(.*)\\[.*]$", "$1[" + branchName + "]");
```

File: src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -11,6 +11,7 @@
     ]]></description>
 
     <change-notes><![CDATA[
+      Version 0.6.0 - Improve stability with IntelliJ 2019.3.3+, Subversion support.
```
Comment (tom-haines) [inline src/main/resources/META-INF/plugin.xml]: I noticed the line breaks for the change notes do not carry though to the jetbrains notes.  Do other like plugins include HTML based breaks ?

**Review by tom-haines — CHANGES_REQUESTED** [review]
A few minor improvements requested. 
Great to see the use of the built in IntelliJ source code plugins!

**Comment by tom-haines** [comment]
Going forward, all changes should apply directly to the wisetime-io owned repo please.

File: src/main/resources/META-INF/plugin.xml (hunk 3 above)
Comment (sukolenvo) [reply src/main/resources/META-INF/plugin.xml]: Yes, release notes have html formatting according to https://intellij-support.jetbrains.com/hc/en-us/community/posts/206752065-Recent-change-notes-formatting-

I will add \<br>

**Review by tom-haines — APPROVED** [review]

**Comment by sukolenvo** [comment]
Merged to wisetime-io repository (https://github.com/wisetime-io/intellij-branch-in-window-title/pull/1). Closing this PR.

**Pull request closed without merging**